common/src/main/java/com/tc/async/impl/AbstractStageQueueImpl.java

```diff
@@ -4,24 +4,51 @@
 
 import com.tc.async.api.EventHandler;
 import com.tc.async.api.EventHandlerException;
+import com.tc.async.api.Sink;
 import com.tc.async.api.Source;
 import com.tc.async.api.SpecializedEventContext;
 import com.tc.async.api.StageQueueStats;
-import com.tc.exception.TCRuntimeException;
-import com.tc.util.UpdatableFixedHeap;
-
-import java.util.concurrent.BlockingQueue;
+import com.tc.logging.TCLoggerProvider;
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
  * @author cschanck
  **/
-public class AbstractStageQueueImpl<EC> {
-
-
-  public AbstractStageQueueImpl() {
+public abstract class AbstractStageQueueImpl<EC> implements StageQueue<EC> {
+
+  private volatile boolean closed = false;
+  final Logger logger;
+  final String stageName;
+  
+  public AbstractStageQueueImpl(TCLoggerProvider loggerProvider, String stageName) {
+    this.logger = loggerProvider.getLogger(Sink.class.getName() + ": " + stageName);
+    this.stageName = stageName;
+  }
+  
+  abstract SourceQueue[] getSources();
+  
+  Logger getLogger() {
+    return logger;
+  }
+  
+  boolean isClosed() {
+    return closed;
   }
 
+  @Override
+  public void setClosed(boolean closed) {
+    this.closed = closed;
+    if (closed) {
```

Comment:
It can't happen because StageImpl protects it.  It's  seems like a bad interdependency between the stage and the queue but i'm not sure how to fix it.  Maybe assert there are no double closes or opens